src/libraries/System.Runtime.InteropServices/gen/Microsoft.Interop.SourceGeneration/MarshallingAttributeInfo.cs

```diff
@@ -593,17 +594,34 @@ private CountInfo CreateCountInfo(AttributeData marshalUsingData, ImmutableHashS
 
                 int maxIndirectionDepthUsedLocal = maxIndirectionDepthUsed;
                 Func<ITypeSymbol, MarshallingInfo> getMarshallingInfoForElement = (ITypeSymbol elementType) => GetMarshallingInfo(elementType, new Dictionary<int, AttributeData>(), 1, ImmutableHashSet<string>.Empty, ref maxIndirectionDepthUsedLocal);
-                if (ManualTypeMarshallingHelper.TryGetLinearCollectionMarshallersFromEntryType(entryPointType, type, _compilation, getMarshallingInfoForElement, out CustomTypeMarshallers? marshallers))
+                if (ManualTypeMarshallingHelper.TryGetLinearCollectionMarshallersFromEntryType(entryPointType, type, _compilation, getMarshallingInfoForElement, out CustomTypeMarshallers? collectionMarshallers))
                 {
                     maxIndirectionDepthUsed = maxIndirectionDepthUsedLocal;
                     return new NativeLinearCollectionMarshallingInfo(
                         entryPointTypeInfo,
-                        marshallers.Value,
+                        collectionMarshallers.Value,
                         parsedCountInfo,
                         ManagedTypeInfo.CreateTypeInfoForTypeSymbol(entryPointType.TypeParameters.Last()));
                 }
+                return NoMarshallingInfo.Instance;
+            }
+
+            if (type is INamedTypeSymbol namedManagedType && entryPointType.IsUnboundGenericType)
+            {
+                if (!ManualTypeMarshallingHelper.TryResolveEntryPointType(
```

Comment:
Add some unit tests for the generator to verify we properly resolve more complex nested generics now with this improved handling?